pinot-broker/src/main/java/org/apache/pinot/broker/queryquota/HelixExternalViewBasedQueryQuotaManager.java

```diff
@@ -348,16 +355,38 @@ private synchronized void createOrUpdateDatabaseRateLimiter(List<String> databas
   }
 
   public synchronized void createOrUpdateApplicationRateLimiter(String applicationName) {
-    createOrUpdateApplicationRateLimiter(Collections.singletonList(applicationName));
+    createOrUpdateApplicationRateLimiter(Collections.singletonList(applicationName), DISABLED_APP_QUOTA);
+  }
+
+  public synchronized void createOrUpdateApplicationRateLimiter(String applicationName, double override) {
+    createOrUpdateApplicationRateLimiter(Collections.singletonList(applicationName), override);
   }
 
   // Caller method need not worry about getting lock on _applicationRateLimiterMap
   // as this method will do idempotent updates to the application rate limiters
-  private synchronized void createOrUpdateApplicationRateLimiter(List<String> applicationNames) {
+  private synchronized void createOrUpdateApplicationRateLimiter(List<String> applicationNames, double override) {
```

Comment:
Consider making override a `@Nullable`, and use `null` to represent not override

Reply:
I added more method to hide the override when it's not needed but kept it as primitive because null is not necessarily more readable and would trigger boxing/unboxing.